tensorboard/plugins/mesh_visualizer/summary_test.py

```diff
@@ -0,0 +1,104 @@
+from tensorboard.plugins.mesh_visualizer import plugin_data_pb2
+from tensorboard.plugins.mesh_visualizer import test_utils
+from tensorboard.util import test_util
+
+
+@test_util.run_v1_only('requires tf.Session') 
+class MeshSummaryTest(tf.test.TestCase):
+
+  def pb_via_op(self, summary_op):
+    """Parses pb proto."""
+    actual_pbtxt = summary_op.eval()
+    actual_proto = summary_pb2.Summary()
+    actual_proto.ParseFromString(actual_pbtxt)
+    return actual_proto
+
+  def verify_proto(self, proto, name):
+    """Validates proto."""
+    self.assertEqual(3, len(proto.value))
+    self.assertEqual("%s_VERTEX" % name, proto.value[0].tag)
+    self.assertEqual("%s_FACE" % name, proto.value[1].tag)
+    self.assertEqual("%s_COLOR" % name, proto.value[2].tag)
+
+  def test_get_tensor_summary(self):
+    """Tests proper creation of tensor summary with mesh plugin metadata."""
+    name = "my_mesh"
+    display_name = "my_display_name"
+    description = "my mesh is the best of meshes"
+    tensor_data = test_utils.get_random_mesh(100)
+    tensor_summary = summary._get_tensor_summary(
+        name, display_name, description, tensor_data.vertices,
+        plugin_data_pb2.MeshPluginData.VERTEX, "", None)
+    with self.test_session():
+      proto = self.pb_via_op(tensor_summary)
+      self.assertEqual("%s_VERTEX" % name, proto.value[0].tag)
+      self.assertEqual(metadata.PLUGIN_NAME,
+                       proto.value[0].metadata.plugin_data.plugin_name)
+
+  def test_op(self):
+    """Tests merged summary with different types of data."""
+    name = "my_mesh"
+    tensor_data = test_utils.get_random_mesh(
+        100, add_faces=True, add_colors=True)
+    config_dict = {"foo": 1}
+    tensor_summary = summary.op(
+        name,
+        tensor_data.vertices,
+        faces=tensor_data.faces,
+        colors=tensor_data.colors,
+        config_dict=config_dict)
+    with self.test_session():
+      proto = self.pb_via_op(tensor_summary)
+      self.verify_proto(proto, name)
+      plugin_metadata = metadata.parse_plugin_metadata(
+          proto.value[0].metadata.plugin_data.content)
+      self.assertEqual(
+          json.dumps(config_dict, sort_keys=True), plugin_metadata.json_config)
+
+  def test_pb(self):
+    """Tests merged summary protobuf with different types of data."""
+    name = "my_mesh"
+    tensor_data = test_utils.get_random_mesh(
+        100, add_faces=True, add_colors=True)
+    config_dict = {"foo": 1}
+    proto = summary.pb(
+        name,
+        tensor_data.vertices,
+        faces=tensor_data.faces,
+        colors=tensor_data.colors,
+        config_dict=config_dict)
+    self.verify_proto(proto, name)
+    plugin_metadata = metadata.parse_plugin_metadata(
+        proto.value[0].metadata.plugin_data.content)
+    self.assertEqual(
+        json.dumps(config_dict, sort_keys=True), plugin_metadata.json_config)
+
+
+if __name__ == "__main__":
+  tf.test.main()
+  
```

Comment:
nit: spurious whitespace on this line

Git will generally inform you when your changes introduce whitespace
errors:


```
<stdin>:8: trailing whitespace.
  
warning: 1 line adds whitespace errors.
```

…and you can surely configure your editor to properly save files with a
terminating newline character.

You can also see an error in the GitHub UI:
![image](https://user-images.githubusercontent.com/4317806/57163695-bdedfc80-6da6-11e9-8fdc-b435db41374b.png)